pangeo_forge_orchestrator/meta_types/bakery.py

```diff
@@ -0,0 +1,240 @@
+KnownImplementations = Literal[tuple(list(known_implementations))]  # type: ignore
+
+FARGATE_CLUSTER = "aws.fargate"
+AKS_CLUSTER = "azure.aks"
+clusters = Literal[FARGATE_CLUSTER, AKS_CLUSTER]  # type: ignore
+
+PANGEO_FORGE_BAKERY_DATABASE = (
+    "https://raw.githubusercontent.com/pangeo-forge/bakery-database/main/bakeries.yaml"
+)
+# ensure that secrets are passed as env var names enclosed in curly braces, e.g. `"{MY_AWS_KEY}"`
+env_var_name = constr(regex=r"{(.*?)}")
+# accepts only strings of integer values 0-9
+run_identifier = constr(regex=r"^\d+$")
+# see https://github.com/pangeo-forge/roadmap/pull/34; the "@" delimiter is introduced here
+feedstock_name_with_version = constr(regex=r".*-feedstock@(0|[1-9]\d*)\.(0|[1-9]\d*)$")
+# TODO: possibly move recipe name validation into `pangeo_forge_orchestrator.meta_types.meta`
+# note that the string prior to the optional ":" delimiter must be a valid python identifier, but
+# the string after ":" is a dictionary key, and therefore does not have this requirement
+recipe_or_dictobj_identifier = constr(
+    regex=r"^([A-Za-z][A-Za-z0-9_]*)([:][A-Za-z][A-Za-z0-9_-]*)?$"
+)
+
+
+class StorageOptions(BaseModel):
+    key: Optional[env_var_name] = None  # type: ignore
+    secret: Optional[env_var_name] = None  # type: ignore
+    anon: Optional[bool] = None
+    # TODO: ensure `v` in `{"client_kwargs": {"headers": {"Authorization": v}}}` is `env_var_name`?
+    client_kwargs: Optional[dict] = None
+    default_cache_type: Optional[s3_default_cache_types] = None
+    default_fill_cache: Optional[bool] = None
+    use_listings_cache: Optional[bool] = None
+    username: Optional[env_var_name] = None  # type: ignore
+    password: Optional[env_var_name] = None  # type: ignore
+
+    class Config:
+        extra = "forbid"
+
+
+@dataclass
+class Endpoint:
+    protocol: KnownImplementations  # type: ignore
+    storage_options: Optional[StorageOptions] = None
+    prefix: Optional[str] = None
+
+
+@dataclass
+class Target:
+    region: regions
+    private: Optional[Endpoint] = None
+    public: Optional[Endpoint] = None
+    description: Optional[str] = None
+    prefix: Optional[str] = None
+
+
+@dataclass
+class FargateClusterOptions:
+    vpc: str
+    cluster_arn: str
+    task_role_arn: str
+    execution_role_arn: str
+    security_groups: List[str]
+
+
+@dataclass
+class Cluster:
+    type: clusters  # type: ignore
+    pangeo_forge_version: str
+    pangeo_notebook_version: str
+    prefect_version: str
+    worker_image: str
+    flow_storage: str
+    flow_storage_protocol: KnownImplementations  # type: ignore
+    flow_storage_options: StorageOptions
+    max_workers: int
+    cluster_options: Optional[FargateClusterOptions] = None
+
+
+@dataclass
+class BakeryMeta:
+    region: regions
+    targets: Dict[str, Target]
+    cluster: Union[Cluster, None]
+    description: Optional[str] = None
+    org_website: Optional[str] = None
```

Comment:
As specified by ADR 4, the bakery needs maintainers / admins too, just like recipes.